packages/mobile/src/screens/artist-coins-explore-screen/ArtistCoinsExploreScreen.tsx

```diff
@@ -126,25 +131,71 @@ export const ArtistCoinsExploreScreen = () => {
   const [sortDirection, setSortDirection] = useState<GetCoinsSortDirectionEnum>(
     GetCoinsSortDirectionEnum.Desc
   )
+  const [offset, setOffset] = useState(0)
+  const [allCoins, setAllCoins] = useState<Coin[]>([])
+  const [hasMore, setHasMore] = useState(true)
+  const [isLoadingNewSearch, setIsLoadingNewSearch] = useState(false)
 
   // Set status bar to light content for dark header
   useStatusBarStyle('light-content')
 
   // Debounce search value to avoid excessive API calls
-  useDebounce(() => setDebouncedSearchValue(searchValue), 300, [searchValue])
+  useDebounce(
+    () => {
+      setDebouncedSearchValue(searchValue)
+      setIsLoadingNewSearch(false)
+    },
+    300,
+    [searchValue]
+  )
 
-  const { data: coinsData, isPending } = useArtistCoins({
+  const {
+    data: coinsData,
+    isPending,
+    isFetching
+  } = useArtistCoins({
```

Comment:
little confused here, shouldnt we make this an infiniteQuery so it can handle it's own limit + offset? ill go ahead and make that change on this branch.